services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/attachment/application/ApplicationAttachmentService.kt

```diff
@@ -33,7 +35,54 @@ class ApplicationAttachmentService(
     private val applicationRepository: ApplicationRepository,
     private val attachmentContentService: ApplicationAttachmentContentService,
     private val scanClient: FileScanClient,
+    private val lockService: LockService,
+    private val contentTransferService: ApplicationAttachmentContentTransferService,
 ) {
+
+    internal val lockName = "applicationAttachmentBlobTransfer"
+
+    @Scheduled(
+        fixedDelayString = "\${haitaton.application.attachments.blobTransferIntervalMilliSeconds}",
+        initialDelayString =
+            "\${haitaton.application.attachments.blobTransferInitialDelayMilliSeconds}"
+    )
+    fun transferAttachmentContentToBlobStorage() {
+        logger.info(
+            "Trying to obtain lock $lockName to start transferring application attachment content from database to Azure Blob Storage."
+        )
+        lockService.doIfUnlocked(lockName) { transferAttachmentContent() }
+    }
+
+    private fun transferAttachmentContent() {
+        logger.info {
+            "Starting to transfer application attachment content from database to Azure Blob Storage"
+        }
+        contentTransferService.nextTransferableAttachment()?.let { attachmentEntity ->
```

Comment:
This would be easier to read if everything inside the let was in a function, so this would be:
```kotlin
        contentTransferService.nextTransferableAttachment()?.let { migrate(it) }
            ?: logger.info { "All application attachment content has been migrated." }
```
Having the Elvis operator after a long block makes it harder to understand what the null condition refers to.

Reply:
Thank you for pointing this out. I hope there are no such things any more.